_data-prepper/pipelines/configuration/processors/anomaly-detector.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: default
+title: Anomaly detector
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# Anomaly detector
+
+## Overview
+
+The `Anomaly detector` processor takes structured data and runs anomaly detection algorithms on fields you can configure in the data. The data must be either an integer or real number in order for the the anomaly detection algorithm to detect anomalies. We recommend that you deploy the `Aggregate` processor in a pipeline before the `Anomaly detector` processor to achieve the best results.  This is because the `Aggregate` processor aggregates events with same keys onto the same host.
+
+This processor uses the `random_cut_forest` mode to detect anomalies. 
+
+
+## Configuration
+
+Configuration for this processor involves specifying a key and specifying options for the mode. You can use the following options to configure the `Anomaly detector` processor.
+
+| Name | Required | Description |
+| :--- | :--- | :--- |
+| `keys` | Yes | A non-ordered `List<String>` which are used as inputs to the ML algorithm to detect anomalies in the values of the keys in the list. At least one key is required.
+| `mode` | Yes |  The ML algorithm (or model) to use to detect anomalies. One of the existing [Modes](#modes) must be provided. See [random_cut_forest](#random_cut_forest).
+
+### Keys
+
+
+<!--- Make the following into a table.--->
+
+### random_cut_forest mode
+* `random_cut_forest`: Processes events using Random Cut Forest ML algorithm to detect anomalies.
+  * After passing a bunch of events with `latency` value between 0.2 and 0.3 are passed through the anomaly detector, when an event with `latency` value 11.5 is sent, the following anomaly event will be generated
+  * More details about this can be found at https://docs.aws.amazon.com/sagemaker/latest/dg/randomcutforest.html
+        ```json
+            { "latency": 11.5, "deviation_from_expected":[10.469302736820003],"grade":1.0}
+        ```
+        Where `deviation_from_expected` is a list of deviations for each of the keys from their corresponding expected values and `grade` is the anomaly grade indicating the severity of the anomaly
+
```

Comment:
@dlvenable @asifsmohammed Are there any other modes that exist, or is this the only one?

Reply:
Currently, there is only one mode. We do this way to allow supporting other modes in the future.